From e5b638125588131c6e5932588c511fb08e8b1772 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 22 Feb 2019 13:53:21 +0100
Subject: [PATCH 1030/1051] net: phy: add support for 5GBASE-KR PHY connection
 types

The 5GBASE-KR is a single lane Serdes backplane ehtenet. Some PHYs use
it to connect 5G ethernet different than 5GBASE-KR, e.g. 88E2110
meeting 5gbase-t requires single serdes lane to be configured to 5G.

Change-Id: I274c879912ec2732d57d29b3c2e3648c1b7fb66d
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/4403
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 include/linux/phy.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/phy.h b/include/linux/phy.h
index 63e87a55702b..b3736dbe62f0 100644
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -89,6 +89,8 @@ typedef enum {
 	PHY_INTERFACE_MODE_XAUI,
 	/* 10GBASE-KR, XFI, SFI - single lane 10G Serdes */
 	PHY_INTERFACE_MODE_10GKR,
+	/* 5GBASE-KR - Single lane 5G Serdes */
+	PHY_INTERFACE_MODE_5GKR,
 	PHY_INTERFACE_MODE_MAX,
 } phy_interface_t;
 
@@ -161,6 +163,8 @@ static inline const char *phy_modes(phy_interface_t interface)
 		return "xaui";
 	case PHY_INTERFACE_MODE_10GKR:
 		return "10gbase-kr";
+	case PHY_INTERFACE_MODE_5GKR:
+		return "5gbase-kr";
 	default:
 		return "unknown";
 	}
-- 
2.17.1

